### ISSUE TITLE
segments not loaded/available on default setup

### ISSUE THREAD

**Issue by andreas-mennel** [romkatv/powerlevel10k#2498]
I started with a clean ubuntu 22.0 in WSL 2

Installed zsh via: `sudo apt-get install zsh`

installed omz via: `sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"`

installed and configured fonts for windows terminal (all icon prompts when executin p10k confiugure can be answered with "yes this looks like ...")

install p10k like: `git clone --depth=1 https://github.com/romkatv/powerlevel10k.git ${ZSH_CUSTOM:-$HOME/.oh-my-zsh/custom}/themes/powerlevel10k`

then updated the Theme referenced in `~/.zshrc`

restartin the shell, execute p10k setup, answer all with yes and choose `prompt style 'Lean'`

Setup finishes, style is applied and everything seems to work. However, if I now type:
`timew` or `azure` I always get `zsh: command not found: timew` or `zsh: command not found: azure`

Any hints on what I'm missing?

**Comment by agyss** [comment]
It's after pressing enter.

**Comment by romkatv** [comment]
Does anything change if you replace powerlevel10k with a different theme?

**Comment by andreas-mennel** [comment]
(image modified layover on system name)
![image](https://github.com/romkatv/powerlevel10k/assets/150674916/661822c3-c500-4f5d-9c70-763c84bb228b)

This is how it looks if I set the theme to random - so it changes the theme (if this was what the question was targeting) but it does not change the behavior if I enter azure.

**Comment by andreas-mennel** [comment]
normal view is like this:
![image](https://github.com/romkatv/powerlevel10k/assets/150674916/4a9ac53b-01b8-4a6b-ae48-370c559dce34)

**Comment by andreas-mennel** [comment]
Just noticed (but may be another bug?) that shrinking the window size leads to this:
![image](https://github.com/romkatv/powerlevel10k/assets/150674916/d60cb6e4-6604-47c6-8768-26c860d368c1)

**Comment by andreas-mennel** [comment]
windows version is Win 11 Enterprise, Version: 10.0.22.621 Build 22621

**Comment by andreas-mennel** [comment]
I'm quite new to zsh/omz/p10k so I may very likely miss something, but I understand this section: https://github.com/romkatv/powerlevel10k#batteries-included

as "If you install p10k, these 'segments' (whatever segments are) are available to you. So if you type `context`  if will show you your username@hostname". Am I just not aware of something (could not find details about these segments or batteries...).

**Comment by andreas-mennel** [comment]
Ok I got this functionality completely wrong - thank you for your patience and assistance...

## What I learned
```
Segments are kind of extensions (additional info shown in terminal) depending on which environment one is in and which tools were previously installed.

For example:
The azure segment shows the currently selected subscription.
For this to work, azure cli must be installed by any means - it does not come with p10k.

The information about which program must be installed for a segment to work can be found within the ~/.p10k.zsh file in this section:
![image](https://github.com/romkatv/powerlevel10k/assets/150674916/cd24a651-ed3b-464e-a91b-e9a67177d6e4)

left is the name of the segment, on the right in the comment is (mostly) the program which must be installed for the segment to work.
```

I may have missed something, but for newcomers this is not obvious and a place to lookup the program per segment outside of the config file would be nice.

**Comment by romkatv** [comment]
Correct, people often use tools to manipulate files in certain formats. The information displayed by "azure" prompt segment is usually managed by `az`.

Not all segments have clearly identifiable associated tools. For example, `dir` or `time` don't have any related tools, while `kubecontext` has over a dozen.

This is usually unnecessary to know for powerlevel10k users. The prompt won't show information that doesn't exist. When it does show something, it's almost always relevant. You can use the information working knowing where it came from.

**Comment by agyss** [comment]
Yep fully agree and also getting it now. My lack in understanding *only* was what exactly segments are - I initially thought about them not *just* being the display part/plugin in p10k, but also a standard tool providing the functionality, which it not is.

So again thanks for clarifying this - it's making way more sense this way as I initally thought about it.

**Comment by romkatv** [comment]
There is this FAQ entry: https://github.com/romkatv/powerlevel10k#which-aspects-of-shell-and-terminal-does-powerlevel10k-affect

Perhaps I should add that powerlevel10k does not define new commands you can run (except for `p10k`).
